pkg/cmd/experimental/buildchain/buildchain.go

```diff
@@ -0,0 +1,346 @@
+	Children  []*ImageRepo `json:"children,omitempty"`
+}
+
+// String helps in dumping a tree in AST format
+func (root *ImageRepo) String() string {
+	tree := ""
+	tree += root.Name
+
+	for _, n := range root.Children {
+		tree += "(" + n.String() + ")"
+	}
+
+	return tree
+}
+
+// Edge represents a build configuration relationship
+// between two nodes
+//
+// Note that this type has no relation with the dot.Edge
+// type
+type Edge struct {
+	Name string `json:"buildConfig"`
+	To   string `json:"to"`
+}
+
+// NewEdge adds a new edge on a parent node
+//
+// Note that this function has no relation
+// with the dot.Edge type
+func NewEdge(name, to string) *Edge {
+	return &Edge{
+		Name: name,
+		To:   to,
+	}
+}
+
+// invalidRepoTagErr is returned when an invalid image repository and tag
+// combination has been passed by the user
+var invalidRepoTagErr = fmt.Errorf("invalid [imageRepository]:[tag] input")
+
+// NewCmdBuildChain implements all the necessary functionality for the build-chain module
+func NewCmdBuildChain(f *clientcmd.Factory, parentName, name string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("%s [image-repository] [OPTIONS]", name),
+		Short: "Output build dependencies of a specific image repository",
+		Long:  longDescription,
+		Run: func(cmd *cobra.Command, args []string) {
+			all := cmdutil.GetFlagBool(cmd, "all")
+			if len(args) > 1 {
+				usageError(cmd, "Must pass nothing, an image repository name, or specify the --all flag")
+			}
+
+			osc, kc, err := f.Clients(cmd)
+			checkErr(err)
+
+			defaultNamespace, err := f.DefaultNamespace(cmd)
+			checkErr(err)
+
+			namespaces := make([]string, 0)
+			if all {
+				nsList, err := kc.Namespaces().List(labels.Everything())
+				checkErr(err)
+				for _, ns := range nsList.Items {
+					namespaces = append(namespaces, ns.Name)
+				}
+			} else {
+				namespaces = append(namespaces, defaultNamespace)
+			}
+
+			// Get all build configurations
+			buildConfigList := make([]buildapi.BuildConfig, 0)
+			for _, namespace := range namespaces {
+				cfgList, err := osc.BuildConfigs(namespace).List(labels.Everything(), labels.Everything())
+				checkErr(err)
+				for _, cfg := range cfgList.Items {
+					buildConfigList = append(buildConfigList, cfg)
+				}
+			}
+
+			// Get all image repositories from build configurations
+			repos := getRepos(buildConfigList, defaultNamespace)
+
+			var tag string
+			if !all && len(args) != 0 {
+				var repo string
+				repo, tag, err = breakDown(args[0])
+				checkErr(err)
+
+				// Validate the specified image repository
+				if _, ok := repos[repo]; !ok {
+					checkErr(fmt.Errorf("invalid image repository: %s", repo))
+				}
+
+				flushMap(repos)
+				repos[repo] = defaultNamespace
+			} else {
+				// Make sure from now on that the --all flag is true
+				// since we are building dependency trees for every
+				// image repository available either in the current
+				// namespace or in every namespace
+				all = true
```

Comment:
It may not be a good idea to assume all==true especially if you have a large repository. Unless they explicitly said that they wanted --all, I would warn that they need to specify an imagerepo